### PR TITLE
Avalanche tutorial contest (Round 2) - Distributed file manager

### PR DESCRIPTION
**Pull request by rajranjan0608**
This is the submission PR (Distributed File Manager) for Round 2 of Avalanche's tutorial contest.

### CODE REVIEW

File: build/tutorials/tutorials-contest/distributed-file-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,1093 @@
...
+Move to the newly created directory and install the basic dependencies.
+
+```bash
+cd dfm-avalanche-react
+npm install --save dotenv web3 @truffle/contract @truffle/hdwallet-provider
+```
+
+Open the file `index.html` file inside of the `public` directory and replace the existing code with the following HTML :
+
+```markup
```
Comment (jpop32) [inline build/tutorials/tutorials-contest/distributed-file-manager/README.md:67]: `html` type will render better.

File: build/tutorials/tutorials-contest/distributed-file-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,1093 @@
...
+
+In a web browser, visit the URL [http://localhost:3000](http://localhost:3000). If npm start has not encountered any errors, we will see the text "Distributed File Manager" at the top of the page as shown in this image :
+
+![](./assets/localhost-frontend.png)
+
+## Create the FileManager contract
+
+Create the file `FileManager.sol` (`.sol` stands for Solidity) inside of the `contracts` directory and paste the following code:
+
+```javascript
```
Comment (jpop32) [inline build/tutorials/tutorials-contest/distributed-file-manager/README.md:280]: `solidity` instead of `javascript`. Check other code blocks too, please.

**Comment by jpop32** [comment]
Please rename the image files so they have a common and unique prefix that can identify them as belonging to this tutorial (GitBook tool we use puts all the images in a single direcory). For example
file-manager-00-name-of-first-image.png,
file-manager-01-name-of-second-image.png,
...

**Comment by rajranjan0608** [comment]
@jpop32 Made required changes. Please have a look. Thanks.

File: build/tutorials/tutorials-contest/distributed-file-manager/frontend/App.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (cgcardona) [inline build/tutorials/tutorials-contest/distributed-file-manager/frontend/App.js:1]: Port js -> ts

File: build/tutorials/tutorials-contest/distributed-file-manager/README.md [hunk 4]
```diff
@@ -0,0 +1,703 @@
...
+## Introduction
+
+In this tutorial we will be making a **Distributed File Manager** using the **IPFS** protocol for storing our files, **Avalanche** network for storing the file references of each address to their uploaded files and **ReactJS** for the frontend code. For compiling and deploying our smart contracts, we will be using **Truffle Suite**.
+
+For your information, [Truffle Suite](https://www.trufflesuite.com) is a toolkit for launching decentralized applications dApps on the EVM. With Truffle you can write and compile smart contracts, build artifacts, run migrations and interact with deployed contracts. This tutorial illustrates how Truffle can be used with the [Avalanche](https://avax.network) network, which is an instance of the EVM.
+
+
+## Prerequisites
+
+- Basic familarity with [Git](https://git-scm.com/), [NodeJS](https://nodejs.org/en) and [npm](https://www.npmjs.com/).
```
Comment (cgcardona) [inline build/tutorials/tutorials-contest/distributed-file-manager/README.md:12]: Replace `-` w/ `*` for list items to match the rest of the docs.

File: build/tutorials/tutorials-contest/distributed-file-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,703 @@
...
+
+  <body>
+    <div id="root"></div>
+  </body>
+</html>
+```
+
+Open the file `App.js` inside of the `src` directory and replace the existing code with the following code:
+
+```javascript
```
Comment (cgcardona) [inline build/tutorials/tutorials-contest/distributed-file-manager/README.md:88]: Use `typescript` instead of `javascript`

**Review by cgcardona — CHANGES_REQUESTED** [review]
Minor changes requested

File: build/tutorials/tutorials-contest/distributed-file-manager/frontend/App.js (hunk 3 above)
Comment (rajranjan0608) [reply build/tutorials/tutorials-contest/distributed-file-manager/frontend/App.js:1]: Issue with porting mentioned [here](https://github.com/ava-labs/avalanche-docs/pull/284#discussion_r707637856)

**Pull request merged**